src/entities/abTrail/ui/AbCanvas.tsx

```diff
@@ -195,24 +196,44 @@ const AbCanvas: FC<Props> = props => {
     return foundNode ?? null;
   };
 
-  const createLogPoint = (point: Point): LogPoint => {
+  const getCurrentAndNextNodeLabels = () => {
     const index = getCurrentIndex();
 
     const currentNode = findNodeByIndex(index);
     const nextNode = findNodeByIndex(index + 1);
 
+    return [currentNode.label, nextNode.label];
+  };
+
+  const createLogPoint = (point: Point): LogPoint => {
+    const [currentNodeLabel, nextNodeLabel] = getCurrentAndNextNodeLabels();
+
     const logPoint: LogPoint = {
       x: point.x,
       y: point.y,
       time: new Date().getTime(),
       valid: null,
-      start: currentNode.label,
-      end: nextNode.label,
+      start: currentNodeLabel,
+      end: nextNodeLabel,
       actual: null,
     };
     return logPoint;
   };
 
+  const createStreamEventPoint = (point: Point): StreamEventPoint => {
+    const [currentNodeLabel, nextNodeLabel] = getCurrentAndNextNodeLabels();
+
+    return {
+      x: (point.x * width) / 100,
+      y: (point.y * width) / 100,
+      time: Date.now(),
+      line_number: logLines?.length - 1,
+      error: StreamEventError.NotDefined,
+      correct_path: `${currentNodeLabel} ~ ${nextNodeLabel}`,
```

Comment:
I'd suggest to move details of creation the string into mapper layer, to make stream-point type more clear with simple primitive properties.